From 6092c10818e9a334c653757f246ad13b0690684c Mon Sep 17 00:00:00 2001
From: Robert Kanter <rkanter@cloudera.com>
Date: Mon, 2 Jun 2014 13:58:01 -0700
Subject: [PATCH 180/596] CLOUDERA-BUILD. CDH-19529 Precommit version of
 MAPREDUCE-5875 Make Counter limits consistent conf
 across JobClient, MRAppMaster, and YarnChild

(cherry picked from commit 7abf3862ee87142935df8be7e58ec020355a8840)
(cherry picked from commit d3812d9d23631598a104e5846df3594f0fa7542c)
---
 .../java/org/apache/hadoop/mapred/JobClient.java   |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobClient.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobClient.java
index fa50ecb..84edfeb 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobClient.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobClient.java
@@ -40,6 +40,7 @@
 import org.apache.hadoop.mapreduce.QueueInfo;
 import org.apache.hadoop.mapreduce.TaskTrackerInfo;
 import org.apache.hadoop.mapreduce.TaskType;
+import org.apache.hadoop.mapreduce.counters.Limits;
 import org.apache.hadoop.mapreduce.filecache.DistributedCache;
 import org.apache.hadoop.mapreduce.security.token.delegation.DelegationTokenIdentifier;
 import org.apache.hadoop.mapreduce.tools.CLI;
@@ -467,6 +468,7 @@ public JobClient(Configuration conf) throws IOException {
    */
   public void init(JobConf conf) throws IOException {
     setConf(conf);
+    Limits.init(conf);
     cluster = new Cluster(conf);
     clientUgi = UserGroupInformation.getCurrentUser();
   }
-- 
1.7.9.5

